test/compilable/test21591.sh

```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
+
+if [ "${OS}" == "windows" ]; then
+    echo "Skipping test21591 on Windows."
+    exit 0
+fi
+
+# Compile both modules separately and verify that the mangled names of the
+# `core.internal.hash.hashOf!(void delegate())` instantiations are equal
+# (in particular, both having the 2nd occurrence of `FZv` backref'd as `Qs`).
+expected="_D4core8internal4hash__T6hashOfTDFZvZQnFNaNbNiNeMxDQs"
```

Comment:
I know this is very brittle and, to make matters worse, also depends on druntime and could thus suddenly break all of a sudden. I'm glad to have remembered the LDC issue with this test case (didn't work with DMD 2.086 at the time either); the problem is that these 2 separately compiled modules are still linkable with DMD (at least back then, haven't tried to link them now), so a simple make-sure-it-links test isn't possible (and could regress in the future with druntime refactorings, i.e., always pass even without this mangling fix).

Reply:
This snippet shows different manglings as link errors:
```
module test;

alias fntype = int function();
void foo1(int function() f1, int function() f2); // _D4test4foo1FPFZiPFZiZv
void foo2(fntype f1, fntype f2);                 // _D4test4foo2FPFZiQeZv

void main() {
    foo1(null, null);
    foo2(null, null);
}
```
Might make a better test case.

Reply:
Oh excellent, thanks!

Reply:
@kinke , I assume you are going to replace this with @rainers suggestion. Please also add a comment with the link to the bugzilla issue.

Reply:
Hmm @rainers:
```D
module test21591;

alias f = void function();
void foo1(f f1, f f2);

void foo2(void function() f1, void function() f2);

static assert(foo1.mangleof == "_D9test215914foo1FPFZvQeZv"); // seems to backref `PFZv`
static assert(foo2.mangleof == "_D9test215914foo2FPFZvPQeZv"); // seems to backref `FZv`
```

Reply:
Okay, should be solved now too, tests extended.